website/docs/guides/best-practices/materializations/materializations-guide-4-incremental-models.md

```diff
@@ -0,0 +1,158 @@
+- 3️⃣  We need **3 key things** in order to accomplish the above:
+  - a **filter** to select just the new or updated records
+  - a **conditional block** that wraps our filter and only applies it when we want it
+  - **configuration** that tells dbt we want to build incrementally and helps apply the conditional filter when needed
+
+Let’s dig into how exactly we can do that in dbt. Let’s say we have an `orders` table that looks like the below:
+
+| order_id | order_status | customer_id | order_item_id | ordered_at | updated_at |
+| -------- | ------------ | ----------- | ------------- | ---------- | ---------- |
+| 123      | shipped      | 7           | 5791          | 2022-01-30 | 2022-01-30 |
+| 234      | confirmed    | 15          | 1643          | 2022-01-31 | 2022-01-31 |
+
+We did our last `dbt build` job on `2022-01-31`, so any new orders since that run won’t appear in our table. When we do our next run (for simplicity let’s say the next day, although for an orders model we’d more realistically run this hourly), we have two options:
+
+- 🏔️ build the table from the **beginning of time again — a _table materialization_**
+  - Simple and solid, if we can afford to do it (in terms of time, compute, and money — which are all directly correlated in a cloud warehouse). It’s the easiest and most accurate option.
+- 🤏 find a way to run **just new and updated rows since our previous run — _an_ _incremental materialization_**
+  - If we _can’t_ realistically afford to run the whole table — due to complex transformations or big source data, it takes too long — then we want to build incrementally. We want to just transform and add the row highlighted in green below, _not_ the previous two that are already in the table, highlighted in yellow.
+
+| order_id | order_status | customer_id | order_item_id | ordered_at | updated_at |
+| -------- | ------------ | ----------- | ------------- | ---------- | ---------- |
+| 123      | shipped      | 7           | 5791          | 2022-01-30 | 2022-01-30 |
+| 234      | confirmed    | 15          | 1643          | 2022-01-31 | 2022-01-31 |
+| 567      | shipped      | 61          | 28            | 2022-02-01 | 2022-02-01 |
+
+### Writing incremental logic
+
+Let’s think through the information we’d need to build such a model that only processes new and updated data. We would need:
+
+- 🕜  **a timestamp indicating when a record was last updated**, let’s call it our `updated_at` timestamp, as that’s a typical convention and what we have in our example above.
+- ⌛ the **most recent timestamp from this table _in our warehouse_** _—_ that is, the one created by the previous run — to act as a cutoff point. We’ll call the model we’re working in`this`, for ‘this model we’re working in’.
+
+That would lets us construct logic like this:
+
+```sql
+select * from orders
+
+where
+  updated_at > (select max(updated_at) from {{ this }})
```

Comment:
I can't find the thread anywhere, but I have heard rumours recently that Snowflake at least doesn't like the undecorated `updated_at` columns and needs `{{ this }}` to be aliased: 

```
  orders.updated_at > (select max(this.updated_at) from {{ this }} as this)
```